console/network/src/testnet3.rs

```diff
@@ -199,14 +200,11 @@ impl Network for Testnet3 {
     }
 
     /// Returns the Varuna universal prover.
-    fn varuna_universal_prover() -> &'static UniversalProver<Self::PairingCurve> {
-        static INSTANCE: OnceCell<UniversalProver<<Console as Environment>::PairingCurve>> = OnceCell::new();
-        INSTANCE.get_or_init(|| {
-            snarkvm_algorithms::polycommit::kzg10::UniversalParams::load()
-                .expect("Failed to load universal SRS (KZG10).")
-                .to_universal_prover()
-                .expect("Failed to convert universal SRS (KZG10) to the prover.")
-        })
+    fn varuna_universal_prover(degree_info: DegreeInfo) -> UniversalProver<Self::PairingCurve> {
```

Comment:
I see that we can no longer keep `UniversalProver` as a `'static`, but what about the `UniversalParams`? those could likely be made `'static` so that we don't need to load them for every call to `ProvingKey::{prove, prove_batch}`

Reply:
Good idea! This would be a somewhat involved change so not sure I can get it in before the code freeze, will tackle in a new PR.

Currently, every time we create a proof, we indeed create `UniversalParams` and grow this according to the circuit size. For a simple transfer it grows to ~ 50MB. But it could grow to gigabytes. If we were to make the object static, we'll also need to add a method to shrink the parameters to our minimum parameter size (~ 10MB) so they don't take up memory forever.